src/ops/key.rs

```diff
@@ -115,57 +113,69 @@ pub fn save_keys(
         }
     }
 
-    match diesel::insert_into(keys::table)
-        .values(keys_to_insert)
-        .execute(&mut db::establish_connection()?)
-    {
-        Ok(num_inserted_rows) => {
-            assert_eq!(num_rows_to_insert, num_inserted_rows);
-            Ok(())
-        }
-        Err(e) => {
-            error!("Failed to save records to db: {:?}", e);
-            Err(Error::internal_server_error())
-        }
+    let pool = &db::establish_connection().await?;
+    let mut num_inserted_rows: u64 = 0;
+    for key in keys_to_insert{
+        let result = match sqlx::query!(
```

Comment:
*Nitpick:* It seems your IDE is not properly configured RE Rust code formatting. Noticed this due to the missing space after `keys_to_insert` and in similar places below. Can help out with configuring your IDE, if needed.